ports/esp32/machine_pwm.c

```diff
@@ -151,38 +206,47 @@ STATIC void esp32_pwm_init_helper(esp32_pwm_obj_t *self,
     }
     self->channel = channel;
 
+    int timer;
+    int freq = args[ARG_freq].u_int;
+
+    if (freq == -1) {
+        freq = PWFREQ;
```

Comment:
If `freq` is not passed in as an argument (so it's -1) then the frequency should not be reconfigured (that's the existing behaviour of this function).  Eg:
```
pwm = machine.PWM(machine.Pin(0), freq=500)
pwm.init(duty=200) # this should retain the existing freq setting
```

Reply:
I think this can be fixed by:
```
if (chan_timer[self->channel] == -1) {
    freq = PWFREQ;
} else {
    freq =  timers[chan_timer[self->channel]].freq_hz;
}
```